ibis/expr/types/relations.py

```diff
@@ -4176,15 +4173,17 @@ def _resolve_predicates(
     ]
     predicates = flatten_predicate(predicates)
 
-    # _.resolve is actually a non-deferred method, so it won't dispatch to
-    # the matched UnresolvedExistsSubquery.resolve() method
-    # TODO(kszucs): remove Deferred.resolve() method
-    replacement = Call(Attr(_, "resolve"), table.op())
-    resolved_predicates = [
-        pred.replace(p.UnresolvedExistsSubquery >> replacement) for pred in predicates
-    ]
-
-    return resolved_predicates
+    rules = (
+        # turn reductions into table array views so that they can be used as
+        # WHERE t1.`a` = (SELECT max(t1.`a`) AS `Max(a)`
+        p.Reduction >> (lambda _: ops.TableArrayView(_.to_expr().as_table()))
+        |
+        # resolve unresolved exists subqueries to IN subqueries
+        p.UnresolvedExistsSubquery >> (lambda _: _.resolve(table.op()))
+    )
+    # do not apply the rules below the following nodes
+    until = p.Value & ~p.WindowFunction & ~p.TableArrayView & ~p.ExistsSubquery
```

Comment:
Here we either could use a common base class or trait to check against, or rethink how we treat the current `Reduction` nodes. Actually reductions are not *valid* on their own, we usually rewrite them to aggregations, subqueries or window functions. 

Perhaps we should treat them as API-level constructs? That way rewrites could get simpler, but we should still fit them somehow into the `Value` node hierarchy.

Reply:
Why are reductions not valid on their own?

Isn't `t.x.sum()` a perfectly valid query?

Reply:
Ah, got it! Yeah, there's no valid SQL that corresponds to only the aggregation function call: you need a `SELECT`.